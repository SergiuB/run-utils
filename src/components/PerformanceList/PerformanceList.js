
import React, { Component } from 'react';
import R from 'ramda';
import classNames from 'classnames';

import Ripple from '../Ripple';

import { secToTime } from '../../services/conversion';

import './PerformanceList.css';

const isSamePerformanceAs = (p1, p2) => p1.vdot === p2.vdot;

class PerformanceItem extends Component {
  state = {
    cursorPos: {}
  };

  handleClick(e){
    // Get Cursor Position
    let cursorPos = {
      top: e.clientY,
      left: e.clientX,
      // Prevent Component duplicates do ripple effect at the same time
      time: Date.now()
    };
    this.setState({ cursorPos: cursorPos });
  }

  render() {
    const { performance, onClick, selected } = this.props;

    return (
      <div
        className={classNames('performance-list-item', 'Ripple-parent', { selected })}
        onMouseUp={this.handleClick.bind(this)}
        onTouchEnd={this.handleClick.bind(this)}
        onClick={() => onClick(performance)}
        >
        <span className="mui--text-body2">{performance.vdot.toFixed(1)}</span>
        <span >{performance.race.label}</span>
        <span className="mui--text-caption">{secToTime(performance.time)}</span>

        <Ripple cursorPos={this.state.cursorPos} />
      </div>
    );
  }
};


class PerformanceList extends Component {
  render() {
    const { performances, onItemClick, selectedPerformance } = this.props;
    return <div className={'performance-list'}>
      {R.compose(
        R.map(performance => <PerformanceItem
          key={`${performance.race.label}-${performance.vdot}`}
          performance={performance}
          onClick={() => onItemClick(performance)}
          selected={isSamePerformanceAs(selectedPerformance, performance)}
          />
        ),
        R.sort(R.descend(p => p.vdot))
      )(performances)}
    </div>;
  }
}

export default PerformanceList;