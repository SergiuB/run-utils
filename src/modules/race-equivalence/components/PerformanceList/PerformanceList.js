import React, { Component } from 'react';
import R from 'ramda';
import classNames from 'classnames';
import IconButton from 'material-ui/IconButton';
import ContentRemove from 'material-ui/svg-icons/content/remove-circle-outline';

import { pinkA200 } from 'material-ui/styles/colors';

import core from '../../../core';
const { Ripple } = core.components;
const  { secToTime } = core.services.conversion;

import './PerformanceList.css';

const isSamePerformanceAs = (p1, p2) => p1.vdot === p2.vdot;

class PerformanceItem extends Component {
  state = {
    cursorPos: {}
  };

  handleClick(e){
    // Get Cursor Position
    let cursorPos = {
      top: e.clientY,
      left: e.clientX,
      // Prevent Component duplicates do ripple effect at the same time
      time: Date.now()
    };
    this.setState({ cursorPos: cursorPos });
  }

  render() {
    const { performance, onClick, onRemove, selected } = this.props;

    return (
      <div className='performance-list-item-wrapper'>
        <div
          className={classNames('performance-list-item', 'Ripple-parent', { selected })}
          onMouseUp={this.handleClick.bind(this)}
          onTouchEnd={this.handleClick.bind(this)}
          onClick={() => onClick(performance)}
          >
          <span className="mui--text-body2">{performance.vdot.toFixed(1)}</span>
          <span >{performance.race.label}</span>
          <span className="mui--text-caption">{secToTime(performance.time)}</span>

          <Ripple cursorPos={this.state.cursorPos} />
        </div>
        {selected && 
          <IconButton
              className='remove-btn' 
              iconStyle={{ color: pinkA200 }}
              onTouchTap={() => onRemove(performance)}
          >
            <ContentRemove />
          </IconButton>
        }
      </div>
    );
  }
};


class PerformanceList extends Component {
  render() {
    const { performances, onItemClick, onItemRemove, selectedPerformance } = this.props;
    return <div className={'performance-list'}>
      {R.compose(
        R.map(performance => <PerformanceItem
          key={`${performance.race.label}-${performance.vdot}`}
          performance={performance}
          onClick={() => onItemClick(performance)}
          onRemove={() => onItemRemove(performance)}
          selected={isSamePerformanceAs(selectedPerformance, performance)}
          />
        ),
        R.sort(R.descend(p => p.vdot))
      )(performances)}
    </div>;
  }
}

export default PerformanceList;