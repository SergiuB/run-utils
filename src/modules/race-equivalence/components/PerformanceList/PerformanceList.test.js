import React from 'react';
import renderer from 'react-test-renderer';
import PerformanceList from './PerformanceList';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

describe('PerformanceList (Snapshot)', () => {
  it('renders correctly with default props', () => {
    const component = renderer.create(<PerformanceList />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('renders correctly with list of performances and selected performance', () => {
    const component = renderer.create(
      <MuiThemeProvider>
        <PerformanceList
          performances={[{
            race: {
              distance: 21.0975, 
              label: "Half",
            },
            time: 6230.93078,
            vdot: 43.30983337421054,
          }, {
            race: {
              distance: 10,
              label: "10K",
            },
            time: 2112.78388,
            vdot: 60.48345269028947,
          }
          ]}
          selectedPerformance={{
            race: {
              distance: 10,
              label: "10K",
            },
            time: 2112.78388,
            vdot: 60.48345269028947,
          }}
      />
    </MuiThemeProvider>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});