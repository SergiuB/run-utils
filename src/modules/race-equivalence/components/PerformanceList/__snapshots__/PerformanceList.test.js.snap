exports[`PerformanceList (Snapshot) renders correctly with default props 1`] = `
<div
  className="performance-list" />
`;

exports[`PerformanceList (Snapshot) renders correctly with list of performances and selected performance 1`] = `
<div
  className="performance-list">
  <div
    className="performance-list-item-wrapper">
    <div
      className="performance-list-item Ripple-parent selected"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}>
      <span
        className="mui--text-body2">
        60.5
      </span>
      <span>
        10K
      </span>
      <span
        className="mui--text-caption">
        35:13
      </span>
      <div
        className="Ripple "
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        } />
    </div>
    <button
      className="remove-btn"
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
          "zIndex": 1,
        }
      }
      tabIndex={0}
      type="button">
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#ff4081",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24">
          <path
            d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z" />
        </svg>
      </div>
    </button>
  </div>
  <div
    className="performance-list-item-wrapper">
    <div
      className="performance-list-item Ripple-parent"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}>
      <span
        className="mui--text-body2">
        43.3
      </span>
      <span>
        Half
      </span>
      <span
        className="mui--text-caption">
        1:43:51
      </span>
      <div
        className="Ripple "
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        } />
    </div>
  </div>
</div>
`;
